core/src/main/java/info/rmapproject/core/rmapservice/impl/openrdf/ORMapObjectMgr.java

```diff
@@ -79,16 +79,15 @@ public boolean isRMapType(SesameTriplestore ts, IRI id, IRI typeIRI) throws RMap
 		if (ts==null || id==null || typeIRI==null){
 			throw new RMapException("Null parameter passed");
 		}
-		boolean isCorrectType = false;
 		try {
-			Set<Statement> stmts = ts.getStatements(id, RDF.TYPE, typeIRI, id);
-			if (stmts != null && stmts.size()>0){
-				isCorrectType = true;
+			if (ts.getConnection().size(id)>0) {
+				return ts.getConnection().hasStatement(id, RDF.TYPE, typeIRI, false, id);
```

Comment:
When you ask RMap to do something with the triplestore it opens a connection, does 1-n queries then should close the connection when it's done.  So this code just gets the existing connection, or opens one if it's not already open.  Sometimes there is a transaction around a specific query, so it shouldn't close here in case we need a rollback.  The close logic [c|sh]ould probably be moved to ORMapService and taken out of the api and webapp, but sometimes there is method to my madness so I need to evaluate how this would impact things in the api before changing it for both api and webapp.  I put in issue #124 yesterday after doing this change - it's currently vague as I'm not sure how the Autowiring of the Triplestore affects it, but it could include a review of how connections are managed throughout and see if we can shift it all into ORMapService.  Make sense?

Reply:
OK, I couldn't un-see this. I have moved the finally logic and added a commit.  Can think about autocloseable stuff another day.